beacon_node/execution_layer/src/test_utils/mock_builder.rs

```diff
@@ -39,25 +43,129 @@ pub enum Operation {
     PrevRandao(Hash256),
     BlockNumber(usize),
     Timestamp(usize),
+    WithdrawalsRoot(Hash256),
 }
 
 impl Operation {
-    fn apply(self, bid: &mut BuilderBid) -> Result<(), BlindedBlockProviderError> {
+    fn apply<B: BidStuff>(self, bid: &mut B) -> Result<(), BlindedBlockProviderError> {
         match self {
             Operation::FeeRecipient(fee_recipient) => {
-                bid.header.fee_recipient = to_ssz_rs(&fee_recipient)?
+                *bid.fee_recipient() = to_ssz_rs(&fee_recipient)?
             }
-            Operation::GasLimit(gas_limit) => bid.header.gas_limit = gas_limit as u64,
-            Operation::Value(value) => bid.value = to_ssz_rs(&value)?,
-            Operation::ParentHash(parent_hash) => bid.header.parent_hash = to_ssz_rs(&parent_hash)?,
-            Operation::PrevRandao(prev_randao) => bid.header.prev_randao = to_ssz_rs(&prev_randao)?,
-            Operation::BlockNumber(block_number) => bid.header.block_number = block_number as u64,
-            Operation::Timestamp(timestamp) => bid.header.timestamp = timestamp as u64,
+            Operation::GasLimit(gas_limit) => *bid.gas_limit() = gas_limit as u64,
+            Operation::Value(value) => *bid.value() = to_ssz_rs(&value)?,
+            Operation::ParentHash(parent_hash) => *bid.parent_hash() = to_ssz_rs(&parent_hash)?,
+            Operation::PrevRandao(prev_randao) => *bid.prev_randao() = to_ssz_rs(&prev_randao)?,
+            Operation::BlockNumber(block_number) => *bid.block_number() = block_number as u64,
+            Operation::Timestamp(timestamp) => *bid.timestamp() = timestamp as u64,
+            Operation::WithdrawalsRoot(root) => *bid.withdrawals_root()? = to_ssz_rs(&root)?,
         }
         Ok(())
     }
 }
 
+// contains functions we need for BuilderBids.. not sure what to call this
+pub trait BidStuff {
+    fn fee_recipient(&mut self) -> &mut ExecutionAddress;
+    fn gas_limit(&mut self) -> &mut u64;
+    fn value(&mut self) -> &mut U256;
+    fn parent_hash(&mut self) -> &mut Hash32;
+    fn prev_randao(&mut self) -> &mut Hash32;
+    fn block_number(&mut self) -> &mut u64;
+    fn timestamp(&mut self) -> &mut u64;
+    fn withdrawals_root(&mut self) -> Result<&mut Root, BlindedBlockProviderError>;
+
+    fn sign_builder_message(
+        &mut self,
+        signing_key: &SecretKey,
+        context: &Context,
+    ) -> Result<BlsSignature, Error>;
+
+    fn to_signed_bid(self, signature: BlsSignature) -> SignedBuilderBid;
+}
+
+impl BidStuff for BuilderBid {
+    fn fee_recipient(&mut self) -> &mut ExecutionAddress {
```

Comment:
would you mind renaming these to e.g. `fee_recipient_mut()`

Reply:
done!